hyperjumptech/bookkeeping
# Fix/5 fix reversal amount

# Hello, Merge/Pull Request (PR)  

## PR type  

### what issue this pr address

1. Closing issue #5 
2. Wrong amount when reversal

### how did you implement

1. Transaction amount was 0
2. Fetch proper amount to reverse

## how to test

1. Clone the project and make sure it runs. You might need to setup an sql database and create its tables.
2. Create a journal (ensure you have the correct hmac authorization):

```bash
curl --request POST \
  --url http://localhost:50051/api/v1/journals \
  --header 'Authorization: 11-22-33-put-in-your-hmac-here ' \
  --header 'Content-Type: application/json' \
  --data '{
	"description": "topup budget 2000 sertiva",
	"creator": "insomnia",
	"transactions": [
		{
			"account_number": "100",
			"description": "TOP UP test",
			"alignment": "DEBIT",
			"amount": 22
		},
		{
			"account_number": "111122223333",
			"description": "TOP UP test",
			"alignment": "CREDIT",
			"amount": 22
		}
	]
}'
```

response:

```bash
{
	"message": "OK",
	"status": "SUCCESS",
	"data": "112233445566"
}
```

3. Using the journal_id from step 2 above, reverse it like this:

```bash
curl --request POST \
  --url http://localhost:50051/api/v1/journals/reversal \
  --header 'Authorization:  11-22-33-put-in-your-hmac-here' \
  --header 'Content-Type: application/json' \
  --data '{
	"description": "reverse test",
	"journal_id": "112233445566",
	"creator": "insomnia"
}'

```

Should respond with

```bash
{
	"message": "OK",
	"status": "SUCCESS",
	"data": "aabbccdd"
}
```

4. You can check the reversal worked properly by checking the balance is reverted.
5. And draw/display the reversal journal id from step 3 to /api/v1/journal/{journal_id}/draw like this: 

```
curl --request GET \
  --url http://localhost:50051/api/v1/journals/aabbccdd/draw \
  --header 'Authorization:   11-22-33-put-in-your-hmac-here ' \
  --header 'Content-Type: application/json' \
```
